File: matrix.py
import random
import time
import psutil
import os

matrix_sizes = [16, 128, 1024]

for n in matrix_sizes:
    print(f"\nMatrix size: {n}x{n}")

    A = [[random.random() for _ in range(n)] for _ in range(n)]
    B = [[random.random() for _ in range(n)] for _ in range(n)]
    C = [[0 for _ in range(n)] for _ in range(n)]

    start = time.time()

    for i in range(n):
        for j in range(n):
            for k in range(n):
                C[i][j] += A[i][k] * B[k][j]

    end = time.time()
    print("Execution Time: %.6f seconds" % (end - start))

    process = psutil.Process(os.getpid())
    memory_usage = process.memory_info().rss / (1024 * 1024)  # Convert to MB
    print(f"Memory Usage: {memory_usage:.2f} MB")

    cpu_times = process.cpu_times()
    print(f"User CPU Time: {cpu_times.user:.6f} seconds")
    print(f"System CPU Time: {cpu_times.system:.6f} seconds")

File: Matrix.java
import java.util.Random;
import java.lang.management.ManagementFactory;
import java.lang.management.ThreadMXBean;

public class Matrix2 {
    static int n = 1024;
    static double[][] a = new double[n][n];
    static double[][] b = new double[n][n];
    static double[][] c = new double[n][n];

    public static void main(String[] args) {

        int[] sizes = {16, 128, 1024}; 

        for (int n : sizes) {

            Random random = new Random();
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    a[i][j] = random.nextDouble();
                    b[i][j] = random.nextDouble();
                    c[i][j] = 0;
                }
            }

            long start = System.currentTimeMillis();

            for (int i = 0; i < n; i++) {
                for (int j = 0; j < n; j++) {
                    for (int k = 0; k < n; k++) {
                        c[i][j] += a[i][k] * b[k][j];
                    }
                }
            }

            long stop = System.currentTimeMillis();
            System.out.println("Execution Time: " + (stop - start) * 1e-3 + " seconds");

            Runtime runtime = Runtime.getRuntime();
            long memoryUsed = (runtime.totalMemory() - runtime.freeMemory()) / (1024 * 1024);  // Convert to MB
            System.out.println("Memory Usage: " + memoryUsed + " MB");

            long cpuTime = ManagementFactory.getThreadMXBean().getCurrentThreadCpuTime() / 1_000_000;
            System.out.println("CPU Time: " + cpuTime + " milliseconds");
    
        }
    }
}